Source/MQTTnet/Server/MqttRetainedMessagesManager.cs

```diff
@@ -21,7 +21,11 @@ public Task Start(IMqttServerOptions options, IMqttNetChildLogger logger)
             if (logger == null) throw new ArgumentNullException(nameof(logger));
             _logger = logger.CreateChildLogger(nameof(MqttRetainedMessagesManager));
             _options = options ?? throw new ArgumentNullException(nameof(options));
+#if NET452
+            return new Task(() => {});
```

Comment:
I had to change this, because the develop branch didn't compile on my machine.

Task.FromResult(0) also just creates a new Task with the result value of 0. Therefore i think it doesn't really makes any difference. But in order to stay consistent I'm going to change it!

Reply:
In general I would use [Task.CompletedTask](https://docs.microsoft.com/en-us/dotnet/api/system.threading.tasks.task.completedtask), and for .NET versions where it is not available (i.e. before 4.6) just use a cached task like `private readonly static Task s_completed = Task.FromResult(0);` so it safes the allocation of a fresh task-object each time.

I can submit a PR to change all uses of `Task.FromResult(0)` to make it allocation free if you want.